sound/soc/codecs/rt711.c

```diff
@@ -457,17 +457,21 @@ static int rt711_set_jack_detect(struct snd_soc_component *component,
 	struct snd_soc_jack *hs_jack, void *data)
 {
 	struct rt711_priv *rt711 = snd_soc_component_get_drvdata(component);
+	int ret;
 
-	rt711->hs_jack = hs_jack;
-
-	if (!rt711->hw_init) {
-		dev_dbg(&rt711->slave->dev,
-			"%s hw_init not ready yet\n", __func__);
-		return 0;
```

Comment:
Don't we need this test? IMHO, It is meaningless if we initialize jack before the codec is initialized.

Reply:
the pm_runtime_resume_and_get guarantees that the bus is running and the codec initialized.